deadlines/src/main/java/com/palantir/deadlines/ProvidedDeadline.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.deadlines;
+
+import java.time.Duration;
+
+record ProvidedDeadline(Duration value, long timestamp) { }
```

Comment:
Worth making it clear that `timestamp` is based on `System.nanotime()` to avoid confusion.

Duration is a great API, but it's pretty heavy weight for internal state tracking -- it might be a bit more efficient to use a `long` based on nanoseconds for our internal tracking.

Reply:
fixed